src/conn/conn_cache.c

```diff
@@ -46,6 +46,111 @@ __cache_config_abs_to_pct(
     return (0);
 }
 
+/*
+ * __validate_cache_config --
+ *     Validate trigger and target values of given configs.
+ */
+static int
+__validate_cache_config(WT_SESSION_IMPL *session, const char *cfg[], bool create)
+{
+    WT_CACHE *cache;
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    bool debug_config;
+
+    conn = S2C(session);
+    cache = conn->cache;
+
+    WT_RET(__wt_config_gets(session, cfg, "debug_mode.configuration", &cval));
+    debug_config = cval.val;
```

Comment:
Isn't this already done from `wti_debug_mode_configuration` and, instead of needing the boolean, you can just use `FLD_ISSET(conn, WT_CONN_DEBUG_CONFIGURATION)`? It is better to have one place getting the configuration.

Reply:
The issue is that the debug flags are set after setting the cache. @bzm-jas posted a comment about it in the JIRA ticket [here](https://jira.mongodb.org/browse/WT-12861?focusedCommentId=6395743&focusedId=6395743&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-6395743).
I asked in our team channel if there was any specific reason why we do it in this order. The current code seems to indicate this order is required but I don't understand why and I wonder if this comment was valid before but now now:
```
    /*
     * This function expects the cache to be created so parse this after the rest of the connection
     * is set up.
     */
    WT_ERR(__wti_debug_mode_config(session, cfg));
```

Reply:
I wish we didn't have to read the configuration twice and call `__wti_debug_mode_config` beforehand but if that's the only solution, I'd be happy with that as well!